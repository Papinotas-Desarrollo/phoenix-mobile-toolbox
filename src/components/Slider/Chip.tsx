// @flow
import PropTypes from 'prop-types';
import React, { FunctionComponent } from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  containerChip: {
    padding: 10,
    marginHorizontal: 5,
  },
  selectedChip: {
    borderBottomColor: colors.papinotasOrange,
    borderBottomWidth: 2,
  },
  selectedTextChip: {
    color: colors.slider.active,
    fontWeight: '600',
  },
  textProps: {
    color: colors.slider.inactive,
    fontSize: 14,
  },
});

interface ChipProps {
  name: string;
  selected: boolean;
  onPressSelection: Function;
}

const Chip: FunctionComponent<ChipProps> = ({ id, name, selected, onPressSelection }) => (
  <TouchableOpacity
    onPress={() => onPressSelection(id || name)}
    style={[styles.containerChip, selected ? styles.selectedChip : null]}
    disabled={selected}
  >
    <Text style={[styles.textProps, selected ? styles.selectedTextChip : null]}>
      {name}
    </Text>
  </TouchableOpacity>
);

Chip.propTypes = {
  name: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  onPressSelection: PropTypes.func.isRequired,
};

export default Chip;
