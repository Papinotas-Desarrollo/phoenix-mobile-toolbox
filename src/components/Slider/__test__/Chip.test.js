import { shallow } from 'enzyme';
import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import Chip from '../Chip';

describe('Snapshot test', () => {
  it('With props snapshot', () => {
    const props = {
      name: 'item1',
      selected: false,
      onPressSelection: jest.fn(),
    };
    const tree = renderer.create(<Chip {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Proptype check', () => {
  it('Callback proptypes defined', () => {
    expect(Chip.propTypes.id).toBeDefined();
    expect(Chip.propTypes.name).toBeDefined();
    expect(Chip.propTypes.selected).toBeDefined();
    expect(Chip.propTypes.onPressSelection).toBeDefined();
  });
});
describe('Functionality', () => {
  it('onPressSelection function to be called', () => {
    const props = {
      name: 'item1',
      selected: false,
      onPressSelection: jest.fn(),
    };
    const wrapper = shallow(<Chip {...props} />);
    wrapper
      .find(TouchableOpacity)
      .at(0)
      .props()
      .onPress();
    expect(props.onPressSelection).toHaveBeenCalled();
  });
});
