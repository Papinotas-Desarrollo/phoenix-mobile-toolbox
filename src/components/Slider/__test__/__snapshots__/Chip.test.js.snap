// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test With props snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#A4C5EB",
          "fontSize": 14,
        },
        null,
      ]
    }
  >
    item1
  </Text>
</View>
`;
